src/en/developer-upgrade-charm.md

```diff
@@ -23,7 +24,10 @@ No other factor is used in determining compatibility: configuration settings in
 particular are converted completely naively, such that any settings from the
 original charm that share a name and type are preserved; any incompatible
 settings are removed; and any new settings take defaults as though freshly
-deployed.
+deployed. Settings may be added or updated by specifying the `--config` flag,
+pointing to a YAML-encoded application config file (see
+[Application configuration][charms-config] for general information on
+configuring a charm).
```

Comment:
It is probably worth mentioning that the *new* version of the charm may actually require additional config or settings so it is worth checking the charm's own documentation.